### PR TITLE
DI-529 Scala style guide

### PR DESCRIPTION
**Pull request by iskabs**
Add a note about Scala, highlights of differences from Java.
Update IntelliJ IDEA code style settings to closer match [Scala style](https://docs.scala-lang.org/style/).

### CODE REVIEW

**Comment by iskabs** [comment]
@AndrewStein please take a look

File: java-scala-diff.md [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+```
+```scala
+// in Scala, just the name of a property
+class Foo {
+  def bar: Int = ...
+
+  def bar_=(bar: Int) { ... }
+}
+
+// if getter/setter are backed by a field, underscore prefix to disambiguate
```
Comment (bradcupit) [inline java-scala-diff.md]: should we make any mention that we'd only back it by a field when necessary? Meaning, we don't want someone reading this and thinking that's the right way to do it all the time.

File: java-scala-diff.md [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+b = new int[] { 5, 6 };
+```
+```scala
+// in Scala, separate by spaces
+tuples.filter { case (s1, s2) => s1 == s2 }
+```
+```java
+// in Java, opening brace may be on next line
+void foo()
+{
```
Comment (bradcupit) [inline java-scala-diff.md]: This is legal Java, but our typical Java code puts `{` on the same line, unless the method has so many parameters/thrown exceptions that the line wraps.

File: java-scala-diff.md [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+// in Scala, opening brace on same line, otherwise compiler might infer semicolon unexpectedly
+def foo(): Unit = {
+  ...
+}
+```
+```java
+// in Java, always use with if/else/for/do/while
+if (foo) {
+  doSomething();
+} else {
```
Comment (bradcupit) [inline java-scala-diff.md]: In our Java standard this would be:

```suggestion
}
else {
```

File: java-scala-diff.md [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+  ...
+}
+
+// Scaladoc with asterisks in third column
+
+/** Does something.
+  *
+  * Some more details.
+  */
+def doSomething(): Unit = {
```
Comment (bradcupit) [inline java-scala-diff.md]: Is this saying we support either of these styles?
Should we make a recommendation for one over the other?

**Comment by franklinejoh** [comment]
@iskabs Am curious if you considered http://www.scalastyle.org or something similar

File: java-scala-diff.md (hunk 2 above)
Comment (iskabs) [reply java-scala-diff.md]: Replaced method with class

File: java-scala-diff.md (hunk 3 above)
Comment (iskabs) [reply java-scala-diff.md]: Done

File: java-scala-diff.md (hunk 4 above)
Comment (iskabs) [reply java-scala-diff.md]: Done. 🤷‍♀️

Comment (bradcupit) [reply java-scala-diff.md]: 😛 

Seems like we'd want to pick one so we have consistency. Do you see one style more prevalent than the other? If so, let's go with that one and recommend not using the other one.

Comment (iskabs) [reply java-scala-diff.md]: Done

**Pull request merged**